### PR TITLE
Add migrateDBOnStart property under persistence

### LINKED ISSUE #506: Add migrateDBOnStart property under persistence
### Description

Add migrateDBOnStart property under persistence will allow Data Index(DI) and Jobs Service(JS) to individually determine if they should use Flyway migrate the database functionality.
Setting this property to false will result in no database migration.
Conversely, setting this property to true will result in database migration.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by rhkp**
Closes #506 

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: config/manager/controllers_cfg.yaml [hunk 1]
```diff
@@ -8,10 +8,10 @@ kanikoDefaultWarmerImageTag: gcr.io/kaniko-project/warmer:v1.9.0
 # Default image used internally by the Operator Managed Kaniko builder to create the executor pods
 kanikoExecutorImageTag: gcr.io/kaniko-project/executor:v1.9.0
 # The Jobs Service image to use, if empty the operator will use the default Apache Community one based on the current operator's version
-jobsServicePostgreSQLImageTag: ""
+jobsServicePostgreSQLImageTag: "apache/incubator-kie-kogito-jobs-service-postgresql:weekly-latest"
 jobsServiceEphemeralImageTag: ""
 # The Data Index image to use, if empty the operator will use the default Apache Community one based on the current operator's version
-dataIndexPostgreSQLImageTag: ""
+dataIndexPostgreSQLImageTag: "apache/incubator-kie-kogito-data-index-postgresql:weekly-latest"
```
Comment (ricardozanini) [inline config/manager/controllers_cfg.yaml:14]: Can you please remove this default value?

**Comment by ricardozanini** [comment]
Can you please take a look on CI failures?

File: config/manager/controllers_cfg.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ kanikoDefaultWarmerImageTag: gcr.io/kaniko-project/warmer:v1.9.0
 # Default image used internally by the Operator Managed Kaniko builder to create the executor pods
 kanikoExecutorImageTag: gcr.io/kaniko-project/executor:v1.9.0
 # The Jobs Service image to use, if empty the operator will use the default Apache Community one based on the current operator's version
-jobsServicePostgreSQLImageTag: ""
+jobsServicePostgreSQLImageTag: "apache/incubator-kie-kogito-jobs-service-postgresql:weekly-latest"
```
Comment (ricardozanini) [inline config/manager/controllers_cfg.yaml:11]: I think we can change this in the const instead of here.

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
@rhkp mind taking a look at the unit tests?

**Review by jakubschwan — APPROVED** [review]

**Pull request merged**